internal/service/vmservice/vm.go

```diff
@@ -393,6 +406,41 @@ func createVM(ctx context.Context, scope *scope.MachineScope) (proxmox.VMCloneRe
 	return res, scope.InfraCluster.PatchObject()
 }
 
+func getVMID(ctx context.Context, scope *scope.MachineScope) (int64, error) {
+	if scope.ProxmoxMachine.Spec.VMIDRange != nil {
+		vmIDRangeStart := scope.ProxmoxMachine.Spec.VMIDRange.Start
+		vmIDRangeEnd := scope.ProxmoxMachine.Spec.VMIDRange.End
+		if vmIDRangeStart != 0 && vmIDRangeEnd != 0 {
+			// Get all used vmids from existing ProxmoxMachines
+			usedVMIDs := []int64{}
+			proxmoxMachines, err := scope.InfraCluster.ListProxmoxMachinesForCluster(ctx)
+			if err != nil {
+				return 0, err
+			}
+			for _, proxmoxMachine := range proxmoxMachines {
+				if proxmoxMachine.GetVirtualMachineID() != -1 {
+					usedVMIDs = append(usedVMIDs, proxmoxMachine.GetVirtualMachineID())
+				}
+			}
+			// Get next free vmid from the range
```

Comment:
@rybnico can you just split the getVMID function, so that sonarcloud don't complain?

Reply:
you can move this section, `// Get next free vmid from the range` to its own function